File: akka-learning-java/src/test/java/org/spiral/java/sql/actor/StringSaveActorTest.java
package org.spiral.java.sql.actor;

import akka.actor.ActorRef;
import akka.actor.Props;
import akka.testkit.TestActorRef;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * ${DESCRIPTION}
 *
 * @author dengguoqing
 * @date 2019/12/12
 * @copyright spiral
 * @since 1.0 Version
 */
public class StringSaveActorTest {

    @Test
    public void createReceive() {
        TestActorRef<StringSaveActor> actorRef = TestActorRef.create(Common.system,
                                                                     Props.create(
                                                                             StringSaveActor.class));
        actorRef.tell(("key"), ActorRef.noSender());

        StringSaveActor akkademyDb = actorRef.underlyingActor();
        assertEquals("key", akkademyDb.string);
        actorRef.tell("value", ActorRef.noSender());
        akkademyDb = actorRef.underlyingActor();
        assertEquals("value", akkademyDb.string);
    }
}

File: akka-learning-scala/src/test/java/org/spiral/scala/sql/actor/StringSaveActorTest.scala
package org.spiral.scala.sql.actor

import akka.actor.ActorSystem
import akka.testkit.TestActorRef
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers

/**
 * ${DESCRIPTION}
 *
 * @author dengguoqing
 * @date 2019/12/12
 * @since 1.0 Version
 * @copyright spiral
 */
class StringSaveActorTest extends AnyFlatSpec with Matchers {
  implicit val system = ActorSystem()


  it should "save last send string" in {
    val actorRef = TestActorRef(new StringSaveActor)
    actorRef ! "key"

    val stringSaveActor = actorRef.underlyingActor
    stringSaveActor.message should equal("key")

    actorRef ! "value"
    val stringSaveActor1 = actorRef.underlyingActor
    stringSaveActor1.message should equal("value")

  }
}
